packages/babel-parser/src/parser/expression.js

```diff
@@ -766,6 +766,9 @@ export default class ExpressionParser extends LValParser {
     optional: boolean,
   ): N.Expression {
     const oldMaybeInArrowParameters = this.state.maybeInArrowParameters;
+    const refExpressionErrors = state.maybeAsyncArrow
+      ? new ExpressionErrors()
```

Comment:
Same here. The idea is that less branching improves performance.

Reply:
Yeah you right. When I step back a bit I found that it's pretty obvious ahaha ^^ Thanks.